internal/resources/azure/linux_virtual_machine.go

```diff
@@ -0,0 +1,103 @@
+package azure
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/infracost/infracost/internal/usage"
+
+	"fmt"
+	"strings"
+
+	"github.com/shopspring/decimal"
+)
+
+type LinuxVirtualMachine struct {
+	Address         string
+	Region          string
+	Size            string
+	UltraSSDEnabled bool
+	OSDiskData      *ManagedDiskData
+	OSDisk          *OSDiskUsage `infracost_usage:"os_disk"`
+	MonthlyHrs      *float64     `infracost_usage:"monthly_hrs"`
+}
+
+var LinuxVirtualMachineUsageSchema = []*schema.UsageItem{
+	{Key: "montly_hrs", ValueType: schema.Float64, DefaultValue: 0},
+	{
+		Key:          "os_disk",
+		ValueType:    schema.SubResourceUsage,
+		DefaultValue: &usage.ResourceUsage{Name: "os_disk", Items: OSDiskUsageSchema},
+	},
+}
+
+func (r *LinuxVirtualMachine) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *LinuxVirtualMachine) BuildResource() *schema.Resource {
+	instanceType := r.Size
+
+	costComponents := []*schema.CostComponent{linuxVirtualMachineCostComponent(r.Region, instanceType, r.MonthlyHrs)}
+
+	if r.UltraSSDEnabled {
+		costComponents = append(costComponents, ultraSSDReservationCostComponent(r.Region))
+	}
+
+	subResources := make([]*schema.Resource, 0)
+
+	var monthlyDiskOperations *decimal.Decimal
+	if r.OSDisk.MonthlyDiskOperations != nil {
+		monthlyDiskOperations = decimalPtr(decimal.NewFromInt(*r.OSDisk.MonthlyDiskOperations))
+	}
+
+	if r.OSDiskData != nil {
+		osDisk := osDiskSubResource(r.Region, r.OSDiskData.DiskType, r.OSDiskData.DiskSizeGB, r.OSDiskData.DiskIOPSReadWrite, r.OSDiskData.DiskMBPSReadWrite, monthlyDiskOperations)
+		subResources = append(subResources, osDisk)
+	}
+
+	return &schema.Resource{
+		Name:           r.Address,
+		CostComponents: costComponents,
+		SubResources:   subResources, UsageSchema: LinuxVirtualMachineUsageSchema,
+	}
+}
+
+func linuxVirtualMachineCostComponent(region string, instanceType string, monthlyHours *float64) *schema.CostComponent {
+	purchaseOption := "Consumption"
+	purchaseOptionLabel := "pay as you go"
+
+	productNameRe := "/Virtual Machines .* Series$/"
+	if strings.HasPrefix(instanceType, "Basic_") {
+		productNameRe = "/Virtual Machines .* Series Basic$/"
+	} else if !strings.HasPrefix(instanceType, "Standard_") {
+		instanceType = fmt.Sprintf("Standard_%s", instanceType)
+	}
+
+	qty := decimal.NewFromFloat(730)
```

Comment:
We could use `schema.HourToMonthUnitMultiplier` here.

Reply:
Nice! There were a few other instances in azure that I also fixed.